File: examples/id-token/token_exchange.cs
using System.Dynamic;
using System.Net.Http.Json;
using System.Text;

Console.WriteLine("Exchange an Identity Token for a session");
var identityToken = "REDACTED";
var httpClient = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, "https://customer.mindtouch.us/@app/auth/token/exchange") {
    Content = new StringContent(identityToken,
    Encoding.UTF8,
    "application/json")
};
var response = await httpClient.SendAsync(request);
dynamic userDynamic = await response.Content.ReadFromJsonAsync<ExpandoObject>();
var userJsonDict = new Dictionary<string, object>(userDynamic);
Console.WriteLine("Welcome " + userJsonDict["fullname"] + " where anonymous equals " + userJsonDict["@anonymous"]);

File: examples/id-token/token_exchange.py
import requests

# an identity token jwt provided by the identity provider
identity_token_jwt = "REPLACEME"

# Expert site/domain
domain = 'customer.mindtouch.com'

# keep a session (for repeated requests as this user)
session = requests.Session()

# Exchange identity token for user session using a matching identity provider
response = session.post(
    f'https://{domain}/@app/auth/token/exchange',
    data=identity_token_jwt,
    headers={'Content-Type': 'application/json; charset=utf-8'}
)

if response.status_code == 200:

    # User has been provisioned based on the identity token.
    user = response.json()
    if user['@anonymous'] == 'false':
        print('Welcome ' + user['fullname'])

        # Additional requests as this user is now possible.
    else:
        raise Exception('Token exchanged failed')
else:
    response.raise_for_status()
    print(response.text)
